cowbird/services/impl/geoserver.py

```diff
@@ -50,42 +59,335 @@ def permission_deleted(self, permission):
+                                                   datastore_name=datastore_name,
+                                                   filename=filename)
+        response_code = response.status_code
+        if response_code == 201:
+            LOGGER.info("Shapefile [%s] has been successfully publish by Geoserver.", filename)
+        elif response_code == 401:
+            LOGGER.error("The request has not been applied because it lacks valid authentication credentials.")
+        elif response_code == 500:
+            LOGGER.error(response.text)
+        else:
+            LOGGER.error("There was an error publishing the following shapefile: [%s]", filename)
+
+    def remove_shapefile(self, workspace_name, filename):
+        # type (Geoserver, str, str) -> None
+        """
+        Remove a shapefile from the specified workspace.
+
+        @param workspace_name: Name of the workspace from which the shapefile will be removed
+        @param filename: The shapefile's name, without file extension
+        """
+        datastore_name = self.get_datastore_name(workspace_name)
+        response = self._remove_shapefile_request(workspace_name=workspace_name,
+                                                  datastore_name=datastore_name,
+                                                  filename=filename)
+        response_code = response.status_code
+        if response_code == 200:
+            LOGGER.info("Shapefile [%s] has been successfully publish by Geoserver.", filename)
+        elif response_code == 401:
+            LOGGER.error("The request has not been applied because it lacks valid authentication credentials.")
+        elif response_code == 500:
+            LOGGER.error(response.text)
+        else:
+            LOGGER.error("There was an error publishing the following shapefile: [%s]", filename)
+
+    #
+    # Helper/request functions
+    #
+    # The following requests were built using Geoserver's REST documentation
+    # https://docs.geoserver.org/master/en/user/rest/index.html
+    #
+    # As well as inspired by the following projects:
+    # - https://github.com/gicait/geoserver-rest
+    # - https://github.com/GeoNode/geoserver-restconfig
+    #
+    # While sometimes harder to get working, data payloads where written in json instead of xml as they are easier
+    # to parse and use without external libraries.
+    def _create_workspace_request(self, workspace_name):
+        # type (Geoserver, str) -> Response
+        """
+        Request to create a new workspace.
+
+        @param workspace_name: Name of workspace to be created
+        @return: Request response
+        """
+        request_url = "{}/workspaces/".format(self.api_url)
+        payload = {"workspace": {"name": workspace_name, "isolated": "True"}}
+        request = requests.post(url=request_url, json=payload, auth=self.auth, headers=self.headers)
+        return request
```

Comment:
All code instances like so are confusing because `request` variable will contain a `Response` object. 
They should be renamed to `response`.